apps/dashboard/src/main/java/com/akto/listener/InitializerListener.java

```diff
@@ -1108,6 +1108,29 @@ public void fillCollectionIdArray() {
         }
     }
 
+    public void updateCustomCollections() {
+        List<ApiCollection> apiCollections = ApiCollectionsDao.instance.findAll(new BasicDBObject());
+        for (ApiCollection apiCollection : apiCollections) {
+            if (ApiCollection.Type.API_GROUP.equals(apiCollection.getType())) {
+                ApiCollectionUsers.computeCollectionsForCollectionId(apiCollection.getConditions(),
+                        apiCollection.getId());
+            }
+        }
+    }
+
+    public void setUpUpdateCustomCollections() {
+        scheduler.scheduleAtFixedRate(new Runnable() {
+            public void run() {
+                AccountTask.instance.executeTask(new Consumer<Account>() {
```

Comment:
Shouldn't this be in `BackwardCompatibility` and `createCollection` method instead of a cron?

Reply:
we have time based filters, due to which the APIs in the custom collections would change. Moreover, in case of API deletions and mergers, we would need to fix the APIs. This cron ensures that. The function being called here is wrong. I'll fix that.